hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/io/compress/zlib/BuiltInGzipCompressor.java

```diff
@@ -86,10 +86,6 @@ public int compress(byte[] b, int off, int len) throws IOException {
 
     int compressedBytesWritten = 0;
 
-    if (currentBufLen <= 0) {
-      return compressedBytesWritten;
-    }
-
```

Comment:
Also, found this bug. If we set an empty input to the compress stream, it will cause endless loop.

Reply:
can we add a test case for this?

Reply:
Without removing it, current test will timeout after removing GzipOutputStream.

Reply:
Do we still need another test for it?

Reply:
Which test is failing before removing the condition? so it passes with the wrapper class but fails after?

Also, we can remove the `currentBufLen` variable now since it is no longer used anywhere else.

Reply:
`testGzipCodec` will cause timeout after removing `GzipOutputStream`.

Because of this line:

```java
`codecTest(conf, seed, 0, "org.apache.hadoop.io.compress.GzipCodec");`
```

It writes an empty input to the compress stream. Due to this `currentBufLen` check, `compress` will return 0 endlessly.

Reply:
I think it's still good to have a dedicated test for this edge case. We can use `@Test(timeout=<value>)` to check the timeout.

Reply:
i see. let me add one then.

Reply:
Thanks!

Reply:
Added one test for empty input case.